tools/serialiser_speed.py

```diff
@@ -0,0 +1,58 @@
+import time
+
+import matplotlib.pyplot as plt
+
+from qat.purr.backends.echo import get_default_echo_hardware
+from qat.purr.compiler.builders import InstructionBuilder, SerialiserBackend
+from qat.purr.compiler.runtime import get_builder
+
+
+def get_test_builder(qubits=32, layers=200):
+    """
+    Make some random large builder to be serialised.
+    """
+    hardware = get_default_echo_hardware(qubits)
+
+    builder = get_builder(hardware)
+    for layer in range(layers):
+        for single_q_layer in range(qubits):
+            builder.X(builder.model.get_qubit(single_q_layer))
+        for two_q_layer in range(qubits - 1):
+            builder.ECR(
+                builder.model.get_qubit(two_q_layer),
+                builder.model.get_qubit(two_q_layer + 1),
+            )
+    for qubit in range(qubits):
+        builder.measure(builder.model.get_qubit(qubit))
+    return builder
+
+
+def serialisation_time(builder, backend: SerialiserBackend):
+    """
+    Time isn't the best way to do this, but it's simple for the purposes of this test.
+    """
+    start = time.time()
+    string = builder.serialize(backend=backend)
+    reconstitued = InstructionBuilder.deserialize(string, backend=backend)
+    end = time.time()
+    return end - start
```

Comment:
Not super critical for CPython implementations, but in general you want to run multiple times and avg the last few runs